### PR TITLE
Implement login endpoint based on signup example

### LINKED ISSUE #5: Implement login endpoint based on signup example
1. Implement the Post `/api/v1/auth/login` endpoint.
2. Implement `LoginRequestDto` with fields `username` and `password`.
3. Implement the `login` method in the authService
4. Declare the `findByUsername` method in the user repository.

Hint:

```java
@Repository
public interface UserRepository extends JpaRepository<User, Long> {
    Optional<User> findByUsername(String username);
}
```

 

> AC1: Send response code 200 if there is a user with a specific username and password.
> AC2: Send a 401 response code if a user with a specific username and password does not exist

### PR DESCRIPTION
**Pull request by Opanasenko-Mykhailo**

### CODE REVIEW

File: src/main/java/ua/com/javarush/oleksandr/reddit/redditcloneabstract/service/AuthService.java [hunk 1]
```diff
@@ -22,4 +24,16 @@
...
     }
 
+    public User login(LoginRequest loginRequest) {
+        String username = loginRequest.getUsername();
+        String password = loginRequest.getPassword();
+
+        User user = userRepository.findByUsername(username)
+                .orElseThrow(() -> new IllegalArgumentException("User not found"));
+
+        if (!password.equals(user.getPassword())) {
```
[the remark this answers is not in the document]
Comment (oleksandr-jr) [reply src/main/java/ua/com/javarush/oleksandr/reddit/redditcloneabstract/service/AuthService.java:34]: We can ignore this :)

Comment (oleksandr-jr) [inline src/main/java/ua/com/javarush/oleksandr/reddit/redditcloneabstract/service/AuthService.java:33]: We can ignore this :)

**Review by Y-Maxym — APPROVED** [review]
Great job, but a little advice. A good tone is considered when the DTO object does not go beyond the scope of the Controller

File: src/main/java/ua/com/javarush/oleksandr/reddit/redditcloneabstract/controller/AuthController.java [hunk 2]
```diff
@@ -1,11 +1,10 @@
 package ua.com.javarush.oleksandr.reddit.redditcloneabstract.controller;
 
 import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
```
Comment (Y-Maxym) [inline src/main/java/ua/com/javarush/oleksandr/reddit/redditcloneabstract/controller/AuthController.java:6]: import...

Comment (oleksandr-jr) [reply src/main/java/ua/com/javarush/oleksandr/reddit/redditcloneabstract/controller/AuthController.java:6]: I'm going to disable this rule :)

**Pull request merged**